packages/floating-label/README.md

```diff
@@ -0,0 +1,84 @@
+# mwc-floating-label
+
+> :warning: These components are a work in progress. They are pre-release and should be considered experimental, as they may undergo major changes before release. We are experimenting with alternate architectures and approaches with the goal of allowing us to bring the most correct and optimal implementation of Material components to the widest possible audiences. Visible progress may be slow, as this research is across teams and repositories so is not consistently reflected in commits to this codebase. :warning:
+
+A [Material Components](https://material.io/components/) icon implementation using [Web Components](https://www.webcomponents.org/introduction)
+
+## Getting started
+
+ * The easiest way to try out mwc-floating-label is to use one of these online tools:
+
+    * Runs in all [supported](#supported-browsers) browsers: [StackBlitz](https://stackblitz.com/edit/mwc-floating-label-example?file=index.js), [Glitch](https://glitch.com/edit/#!/mwc-floating-label-example?path=index.html)
+
+    * Runs in browsers with [JavaScript Modules](https://caniuse.com/#search=modules): [JSBin](https://jsbin.com/qobefic/edit?html,output),
+    [CodePen](https://codepen.io/jcrestel/pen/KGWBLd).
+
+* You can also copy <!-- TODO(elephants@google.com):update link -->[this HTML file](https://gist.githubusercontent.com/JCrestel/9ed0acbd4d372a174b89cd6c58457636/raw/eadc711e5c4b89d9de3dea0d89e1d3797e0eaba3/index.html) into a local file and run it in any browser that supports [JavaScript Modules]((https://caniuse.com/#search=modules)).
+
+* When you're ready to use mwc-floating-label in a project, install it via [npm](https://www.npmjs.com/). To run the project in the browser, a module-compatible toolctain is required. We recommend installing the [Polymer CLI](https://github.com/Polymer/polymer-cli) and using its development server as follows.
+
+  1. Ensure the webcomponents polyfills are included in your HTML page
+
+      - Install webcomponents polyfills
+
+          ```npm i @webcomponents/webcomponentsjs```
+
+      - Add webcomponents polyfills to your HTML page
+
+          ```<script src="@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>```
+
+  1. Add mwc-floating-label to your project:
+
+      ```npm i @material/mwc-floating-label```
+
+  1. Import the mwc-floating-label definition and styles into your module script:
+
+      ```ts
+      import {LitElement, html} from "lit-element";
+      import {floatingLabel} from "@material/mwc-floating-label";
+      import {style} from '@material/mwc-floating-label/mwc-floating-label-css.js';
+      ```
+
+  1. Apply the floating label directive.
```

Comment:
Do we actually anticipate users will directly use the floating label directive? It seems like an implementation detail for us. Cool if so, just want to make sure we thought about it carefully -- since it becomes part of our support burden.

If we do want to support using these directly, I'd still suggest adding a really clear explanation towards the top to say that these are internal implementation details, and linking to e.g. the `textarea` README as the thing most users will actually want to use.